Backend data

typeholes opened the pull request:
Moves the data and processing for traces to the extension for better performance.

Also includes the ability to create a mock of the extension panel for ease of development of the UI

Commit:
initial nuxt UI integration

Commit:
type safe message passing

Commit:
added tailwind

Commit:
send trace to UI

Commit:
dev panel to enable testing without vscode

Commit:
restore tsconfig from main

Commit:
fix ci failures

Commit:
ci fixes

Commit:
build ui in test ci task

Commit:
added nuxt-ui

Commit:
goto trace from source location

Commit:
add type counts to diagnostics

typeholes committed:
Merge branch 'main' into nuxt-ui

typeholes committed:
Merge branch 'main' into nuxt-ui

typeholes committed:
fixed type count injection into diagnostics

typeholes committed:
Merge branch 'auto-trace' into nuxt-ui

typeholes committed:
add options for statement level benchmarks, iteration count, and
restarting tsserver

typeholes committed:
lint fixes

typeholes committed:
include non top level statement

typeholes committed:
Merge branch 'statement-level-benchmarks' into nuxt-ui

typeholes committed:
chore: command to gather traces

typeholes committed:
do not run benchmarks when iterations is 0

typeholes committed:
restored newline to package.json (again)

typeholes committed:
feat: diagnostics from trace

danielroe committed:
Merge remote-tracking branch 'origin/main' into nuxt-ui

danielroe committed:
Merge branch 'main' into nuxt-ui

danielroe committed:
chore: build ui in `build` step

danielroe committed:
chore: remove quasar paths

danielroe committed:
chore: update deps

danielroe committed:
Merge branch 'main' into nuxt-ui

typeholes committed:
fix: activation failure and package failure

typeholes committed:
Merge branch 'main' into generate-contributions

typeholes committed:
fix: commands not showing

typeholes committed:
chore: fix vscode types dependency

typeholes committed:
chore: update pnpm lock file

typeholes committed:
feat: send trace files in chunks, perf fixes

typeholes committed:
Merge branch 'main' into perf

typeholes committed:
removed traces from playground as they are no longer needed

typeholes committed:
removed outdated package script

typeholes committed:
reset filters on process trace files

typeholes committed:
chore: vscode command contexts

typeholes committed:
settings for thresholds

typeholes committed:
ability to inject markdown descriptions into config from .md files

typeholes committed:
trace diagnostics respect threshold settings

typeholes committed:
relative trace diagnostics

typeholes committed:
ensure webview is open when running commands that interact with it

typeholes committed:
feedback during trace (WIP)

typeholes committed:
ui progress while trace running

typeholes committed:
ui progress while trace running

typeholes committed:
Merge branch 'main' into UX

typeholes committed:
removed js from playground

typeholes committed:
(WIP)

typeholes committed:
fix: bad performance with many types

typeholes committed:
fix: NaNs and negatives in trace diagnostics

typeholes committed:
fix: error when reopening trace view

typeholes committed:
remove unused imports

typeholes committed:
feat: persistent state

typeholes committed:
formatting

typeholes committed:
trace data and diagnostics working on backend

typeholes committed:
remove data from UI and mock extension for UI dev

typeholes committed:
fix: send trace command from context menu

typeholes committed:
chore: update readme

typeholes committed:
persistent storage moved to backend

typeholes committed:
incremental on demand fetch of trace data in UI

typeholes committed:
fix: off by one for positions

typeholes committed:
fix: go to position in editor

typeholes committed:
fix: trace view should not be forced open

typeholes committed:
chore: build cleanup

typeholes committed:
tracer vscode settings

typeholes committed:
setting to disable metrics from traces

typeholes committed:
feat: status bar

danielroe commented on .vscode/settings.json:
```diff
@@ -6,5 +6,6 @@
     "editor.defaultFormatter": "dbaeumer.vscode-eslint"
-  }
+  },
+  "tsperf.tracer.traceCmd": "/home/hw/projects/TypeScript/bin/tsc --noEmit --generateTrace ${traceDir}"
```
this should probably not be committed

danielroe committed:
chore: small tweaks to docs

danielroe committed:
chore: tidy

danielroe committed:
chore: remove specific attribution

danielroe reviewed (commented):
exciting!

I can't yet figure out how to run the trace though - what am I missing?

https://github.com/tsperf/tracer/assets/28706372/d39d04b0-9178-4127-a3b2-4a591484dadb

typeholes committed:
chore: remove devMessages which are no longer needed

typeholes committed:
fix: single character diagnostic spans to avoid confusing nested spans

typeholes committed:
fix: occasionaly off by one in goto position

typeholes committed:
stream tree to UI to avoid long hang

typeholes commented:
You can run the trace by clicking on the status bar item or running the "Tracer: tsc trace" command

typeholes commented:
fixed rare off by one in the goto position in trace and now streaming the tree start nodes in case there are enough to cause lag when sending all at once.

@ssalbdivad pointed out that overlapping error spans were confusing which highlighted that full spans didn't give the user any indication of what positions were in the trace, so I changed to single character spans for the trace diagnostics

typeholes commented:
Ah you did run the command.  If it fails, the error output is shown in the terminal.   I set the default trace command to use npx, but that obviously doesn't work for all project.  here it should be pnpm.  Maybe I should just default to nothing and prompt the user on first trace run? 
![image](https://github.com/tsperf/tracer/assets/84949734/afb12a37-31b1-48d2-acf1-35b8841b30da)

danielroe committed:
fix: remove localforage

danielroe committed:
fix: do not emit traces

typeholes committed:
launch debug in new profile

typeholes committed:
chore: git ignore playground vscode settings

typeholes committed:
fix: off by one diagnostic position when using non relative thresholds

typeholes committed:
fix: filenames in ui not reset when sending trace directory

typeholes committed:
fix: project name fallback to hash

typeholes commented:
fixes #22
